packages/contracts/contracts/engine/V2/GenArt721Minter_PBAB.sol

```diff
@@ -178,24 +178,38 @@ contract GenArt721Minter_PBAB is ReentrancyGuard {
     /**
      * @notice Purchases a token from project `_projectId`.
      * @param _projectId Project ID to mint a token on.
+     * @param _maxPricePerToken Maximum price of token being allowed by the purchaser, no decimal places. Required if currency is ERC20.
+     * @param _currencyAddress Currency address of token.
      * @return _tokenId Token ID of minted token
      */
     function purchase(
-        uint256 _projectId
+        uint256 _projectId,
```

Comment:
we could add purchase & purchaseTo functions that auto-forward assuming ETH purchases (i.e. _maxPricePerToken set to msg.value, _currencyAddress set to address(0)), making this backwards-compatible for non-ERC20 purchases. WDYT?

Reply:
Great call out! I added purchase and purchaseTo here and in GenArt721MinterBurner_PBAB